functions/src/telegram/webdevitalia.ts

```diff
@@ -1,71 +1,96 @@
+  sendRules(context, context.message?.message_id)
+);
 
-bot.hears(['/dontasktoask', '/nonchiederedichiedere'], (context: TelegrafContext) => {
-  const messageReplyTarget = context.message?.reply_to_message?.message_id ?? context.message?.message_id
+bot.hears(["/contribute", "/contribuisci"], (context: Context) => {
+  return context.reply(
+    `Contribuisci al gruppo: https://github\\.com/insieme-dev/community`,
+    {
+      parse_mode: "MarkdownV2",
+      disable_web_page_preview: true,
+      reply_to_message_id: context.message?.message_id ?? undefined,
+    }
+  );
+});
 
-  const extra: ExtraEditMessage = messageReplyTarget ?
-    Extra.inReplyTo(messageReplyTarget).markdown().webPreview(false).markup(true) :
-    Extra.markdown().webPreview(false).markup(true)
+bot.hears(["/dontasktoask", "/nonchiederedichiedere"], (context: Context) => {
+  const messageReplyTarget =
+    // @ts-ignore - reply_to_message exists but telegraf typings are flawed
+    context.message?.reply_to_message?.message_id ??
+    context.message?.message_id;
 
-  return context.reply(`
+  return context.reply(
+    `
 Leggi questo per favore e poi rielabora la tua domanda:
-  - https://nonchiederedichiedere.com (ITA)
-  - https://dontasktoask.com (ENG)
-`)
-})
+  🇮🇹 https://nonchiederedichiedere\\.com
+  🇺🇸 https://dontasktoask\\.com
+`,
+    {
+      reply_to_message_id: messageReplyTarget,
+      parse_mode: "MarkdownV2",
+      disable_web_page_preview: true,
+    }
+  );
+});
 
-bot.hears(['/rielabora'], async (context: TelegrafContext) => {
-  const {message_id, from} = context.message?.reply_to_message ?? context.message ?? {};
+bot.hears(["/rielabora"], async (context: Context) => {
+  console.log("RIELABORA", context);
+
+  const { message_id, from } =
+    // @ts-ignore - reply_to_message exists but telegraf typings are flawed
+    (context.message?.reply_to_message ?? context.message ?? {}) as Message;
 
   if (from && message_id) {
     const rulesMessage = await sendRules(context);
     await context.deleteMessage(message_id);
 
-    const extra = from.username
-      ? Extra.webPreview(false).markup(true)
-      : Extra.webPreview(false).markdown(true);
-
     const mention = from.username
       ? `@${from.username}`
       : `[${from.first_name}](tg://user?id=${from.id})`;
 
     return context.reply(
       `${mention} leggi le regole e poi rielabora la tua domanda per favore`,
-      extra.inReplyTo(rulesMessage.message_id) as ExtraEditMessage
+      {
+        disable_web_page_preview: true,
+        parse_mode: "MarkdownV2",
+        reply_to_message_id: rulesMessage.message_id,
+      }
     );
   } else {
     return context;
   }
-})
+});
 
-exports.bot = functions
-  .region('europe-west1')
-  .https
-  .onRequest((req, res) => bot.handleUpdate(req.body, res).then(rv => !rv && res.sendStatus(200)))
+export default functions.region("europe-west1").https.onRequest((req, res) => {
+  return bot.handleUpdate(req.body, res);
```

Comment:
prima inviava un setStatus a 200 perché firebase aveva un bug strano. Senza questo mi si bloccava tutto. Possiamo mandarlo online e provarlo ma ho paura sia rimasto.

Reply:
Che bug era? Con setStatus a volte mi crashava l'applicazione, altre volte sparava solo un warning dicendo che cercavo di impostare gli header dopo aver mandato la risposta, quando non era vero...

Reply:
il bug si presentava solo in produzione